connector/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/v2/PostgresIntegrationSuite.scala

```diff
@@ -230,6 +230,14 @@ class PostgresIntegrationSuite extends DockerJDBCIntegrationV2Suite with V2JDBCT
     }
   }
 
+  test("SPARK-49695: Postgres fix xor push-down") {
+    val df = spark.sql("select id, name from employee where id ^ 6 = 0")
```

Comment:
```suggestion
    val df = spark.sql("select id, name from employee where id ^ 6 = 0")
    assert(!df.queryExecution.optimizedPlan.exists { plan =>
      plan.isInstanceOf[LogicalPlan]
    })
```

We should assert xor is actually pushed down here